src/mbgl/algorithm/generate_clip_ids_impl.hpp

```diff
@@ -64,7 +64,7 @@ void ClipIDGenerator::update(std::vector<std::reference_wrapper<Renderable>> ren
         uint8_t count = 1;
         for (auto& it : renderables) {
             auto& renderable = it.get();
-            if (!renderable.used) {
+            if (!renderable.used || !renderable.needsClipping) {
```

Comment:
Any feeling on `!(renderable.used && renderable.needsClipping)` instead? It has the same results as the above expression for all four `bool, bool` permutations.

Reply:
Interesting nit @sansumbrella — I'm curious about your reasoning behind this comment? In general I find code reviews to be most helpful when they express suggestions about improving functionality, efficiency, or readability, rather than simply suggesting a different way to write the same thing (with the same or, theoretically, [slightly worse](https://en.wikipedia.org/wiki/Short-circuit_evaluation) performance) — unless there's some other motivation behind this suggestion, in which case please share such reasoning while reviewing!

Reply:
I personally prefer how the existing code is with `(!renderable.used || !renderable.needsClipping)`. The logic is easier to understand when glancing at it.